site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -48,6 +52,101 @@ private function validateDiffViewerType(string $type) {
         return true;
     }
 
+    
+    /**
+     * Route for randomizing peer assignments with 'One Grades Many'
+     *
+     * @Route("/courses/{_semester}/{_course}/gradeable/{gradeable_id}/ajaxRandomizePeers", methods={"POST"})
+     * @AccessControl(role="INSTRUCTOR")
+     */
+    public function ajaxRandomizePeers($gradeable_id) {
+        /* How does this function work?
+        1 - Number of students to grade (Y) is taken from the client using POST
+            1.1 - If the number is > number of students, then ALL grade ALL.
+        2 - Query DB to get students from registration section(X) (Without taking in students in NULL section)
+        3 - Randomize the order of students
+        4 - Randomly Select Y offsets
+        5 - Shift the random order by the offsets to create the matrix, with no duplicates, and exactly Y assignments and & graders for each student.  no student grades self.
+        */
+        $number_to_grade = $_POST['number_to_grade'];
+        $gradeable = $this->tryGetGradeable($gradeable_id);
+        if ($gradeable === false) {
+            $this->core->addErrorMessage('Invalid Gradeable!');
+            $this->core->redirect($this->core->buildCourseUrl());
+        }
+        $all_grade_all = false;
+        $order = new GradingOrder($this->core, $gradeable, $this->core->getUser(), true);
+        $student_array = [];
+        $student_list = [];
+        $students = $this->core->getQueries()->getUsersByRegistrationSections($order->getSectionNames());
+        foreach ($students as $student) {
+             $reg_sec = ($student->getRegistrationSection() === null) ? 'NULL' : $student->getRegistrationSection();
+             $sorted_students[$reg_sec][] = $student;
+             array_push($student_list, ['user_id' => $student->getId()]);
+             array_push($student_array, $student->getId());
+        }
+        $number_of_students = count($student_list);
+        if ($number_to_grade > $number_of_students) {
+            $all_grade_all = true;
+        }
+        if ($all_grade_all) {
+            $this->core->getOutput()->renderJSONSuccess("Invalid Number of Students Entered");
```

Comment:
Actually this is the case if the number of students entered are more than number of students in a class. I think this feature was a result of the discussion on how all grade all feature should be. This will enable all grades all and every person in class will grade everyone.